### ISSUE TITLE
error communicating with Plex Server

### ISSUE THREAD

**Issue by larsje993** [Tautulli/Tautulli#1060]
Im having an error communicating with Plex Server.
here is what I did. Migrate PlexPy.db from old Ubuntu server to new Ubuntu server.
After this, it all worked properly. 

Since 2 days there is an error. 
I did fetch the a new token, uncheck SSL, filled in the correct IP and port number.

Don't know what to do next.. 

This is my log file: 

[plexpy_log.txt](https://github.com/JonnyWong16/plexpy/files/1109454/plexpy_log.txt)

**Comment by JonnyWong16** [comment]
https://github.com/JonnyWong16/plexpy/wiki/Frequently-Asked-Questions-(FAQ)#general-q16

If that doesn't work, try removing PlexPy from you list of devices in Plex > Settings > Devices, then fetching a new token.

**Comment by larsje993** [comment]
Thanks for your reply. 
I can successfully acces the hash.plex.direct. 
I already unmounted the PlexPy as a device and then added a new token.. 
Till no luck.

**Comment by JonnyWong16** [comment]
You can try a manual token: https://support.plex.tv/hc/en-us/articles/204059436-Finding-an-authentication-token-X-Plex-Token

This will be a temporary token that will expire after some time though.

**Comment by larsje993** [comment]
Wow lol. Didn't expect that that was going to work. But it seems to work, but its a kind of strange..
Any idea what could be the problem?

**Comment by JonnyWong16** [comment]
Was the token actually changing each time you fetched a new one? Or was Plex just giving you the same token each time?

**Comment by larsje993** [comment]
I did got an new token each time i fetched it.

**Comment by AngryNoodlez** [comment]
Hi,

I'm also receiving this issue after an update to the latest PlexPass version of Plex (v1.7.5.4035)

If I check the Plex Media Server logs, I can see the connection is being refused as it's supposedly not connecting securely from PlexPy.

My Plex is set to Require secure connections, and PlexPy is set to connect via SSL.

What is odd, is if I set Plex to "Prefer Secure Connections" and leave PlexPy set to use SSL - It connects without a problem. A snippit of the Plex Media Server logs below;

    Jun 29, 2017 17:18:31.317 [3408] DEBUG - Beginning read from two-way stream.
    Jun 29, 2017 17:18:32.895 [2596] DEBUG - TrayIcon::WindowProc received DisplayIcon timer message
    Jun 29, 2017 17:18:32.895 [2596] WARN - Shell_NotifyIcon(NIM_ADD) failed: 0x0
    Jun 29, 2017 17:18:32.911 [2744] DEBUG - Request: [10.10.100.27:55352] rejected because secure connections are required
    Jun 29, 2017 17:18:32.927 [2744] DEBUG - Request: [10.10.100.27:55353] rejected because secure connections are required
    Jun 29, 2017 17:18:37.896 [2596] DEBUG - TrayIcon::WindowProc received DisplayIcon timer message
    Jun 29, 2017 17:18:37.896 [2596] WARN - Shell_NotifyIcon(NIM_ADD) failed: 0x0
    Jun 29, 2017 17:18:42.911 [2596] DEBUG - TrayIcon::WindowProc received DisplayIcon timer message
    Jun 29, 2017 17:18:42.911 [2596] WARN - Shell_NotifyIcon(NIM_ADD) failed: 0x0
    Jun 29, 2017 17:18:45.412 [2744] DEBUG - Request: [10.10.100.27:55358] rejected because secure connections are required
    Jun 29, 2017 17:18:45.427 [2744] DEBUG - Request: [10.10.100.27:55359] rejected because secure connections are required
    Jun 29, 2017 17:18:47.896 [2744] DEBUG - Request: [10.10.100.27:55360] rejected because secure connections are required
    Jun 29, 2017 17:18:47.896 [2744] DEBUG - Request: [10.10.100.27:55361] rejected because secure connections are required

**Comment by JonnyWong16** [comment]
Have you read the FAQ answer on SSL connections?

https://github.com/JonnyWong16/plexpy/wiki/Frequently-Asked-Questions-(FAQ)#general-q17

**Comment by AngryNoodlez** [comment]
Hi Jonny,

I have. When originally implementing PlexPy, I had the DNS Rebinding problem with my PfSense and had to include the following parameter on my DNS Resolver;

    server:
    private-domain: "plex.direct"

This works fine for me accessing my server using the hashed URL locally, though to double check I've instead implemented the suggestion from your FAQ and no change unfortunately.

This has been working for the entire time I've had PlexPy running, until the latest PlexPass update of Plex, so I'm a little hesitant to disable forced SSL.

I may just have to roll-back plex.. and hope the next update doesn't suffer the same problem.

~Mat

**Comment by JonnyWong16** [comment]
I need to see your PlexPy logs, not your Plex logs.

**Comment by AngryNoodlez** [comment]
Thanks Jonny,

Here's a snippet, though I can upload full logs if required;

	2017-06-29 17:05:43 - WARNING :: CP Server Thread-12 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:05:43 - WARNING :: CP Server Thread-12 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:05:43 - WARNING :: CP Server Thread-12 : Unable to retrieve data for get_current_activity_header.
	2017-06-29 17:05:57 - WARNING :: CP Server Thread-11 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:05:57 - WARNING :: CP Server Thread-11 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:05:57 - WARNING :: CP Server Thread-11 : Unable to retrieve data for get_activity.
	2017-06-29 17:05:57 - WARNING :: CP Server Thread-8 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:05:57 - WARNING :: CP Server Thread-8 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:05:57 - WARNING :: CP Server Thread-8 : Unable to retrieve data for get_current_activity_header.
	2017-06-29 17:06:13 - WARNING :: CP Server Thread-10 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:06:13 - WARNING :: CP Server Thread-10 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:06:13 - WARNING :: CP Server Thread-10 : Unable to retrieve data for get_activity.
	2017-06-29 17:06:13 - WARNING :: CP Server Thread-9 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:06:13 - WARNING :: CP Server Thread-9 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:06:13 - WARNING :: CP Server Thread-9 : Unable to retrieve data for get_current_activity_header.
	2017-06-29 17:06:26 - WARNING :: Thread-23 : Failed to access uri endpoint /library/recentlyAdded?X-Plex-Container-Start=0&X-Plex-Container-Size=10 with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:06:26 - WARNING :: Thread-23 : PlexPy Pmsconnect :: Unable to parse XML for get_recently_added: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:06:28 - WARNING :: CP Server Thread-7 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:06:28 - WARNING :: CP Server Thread-7 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:06:28 - WARNING :: CP Server Thread-7 : Unable to retrieve data for get_activity.
	2017-06-29 17:06:28 - WARNING :: CP Server Thread-4 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:06:28 - WARNING :: CP Server Thread-4 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:06:28 - WARNING :: CP Server Thread-4 : Unable to retrieve data for get_current_activity_header.
	2017-06-29 17:06:43 - WARNING :: CP Server Thread-3 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:06:43 - WARNING :: CP Server Thread-3 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:06:43 - WARNING :: CP Server Thread-3 : Unable to retrieve data for get_activity.
	2017-06-29 17:06:43 - WARNING :: CP Server Thread-5 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:06:43 - WARNING :: CP Server Thread-5 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:06:43 - WARNING :: CP Server Thread-5 : Unable to retrieve data for get_current_activity_header.
	2017-06-29 17:06:58 - WARNING :: CP Server Thread-6 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:06:58 - WARNING :: CP Server Thread-6 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:06:58 - WARNING :: CP Server Thread-6 : Unable to retrieve data for get_activity.
	2017-06-29 17:06:58 - WARNING :: CP Server Thread-12 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:06:58 - WARNING :: CP Server Thread-12 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:06:58 - WARNING :: CP Server Thread-12 : Unable to retrieve data for get_current_activity_header.
	2017-06-29 17:07:13 - WARNING :: CP Server Thread-11 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:07:13 - WARNING :: CP Server Thread-11 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:07:13 - WARNING :: CP Server Thread-11 : Unable to retrieve data for get_activity.
	2017-06-29 17:07:13 - WARNING :: CP Server Thread-8 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:07:13 - WARNING :: CP Server Thread-8 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:07:13 - WARNING :: CP Server Thread-8 : Unable to retrieve data for get_current_activity_header.
	2017-06-29 17:07:26 - WARNING :: Thread-21 : Failed to access uri endpoint /library/recentlyAdded?X-Plex-Container-Start=0&X-Plex-Container-Size=10 with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:07:26 - WARNING :: Thread-21 : PlexPy Pmsconnect :: Unable to parse XML for get_recently_added: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:07:28 - WARNING :: CP Server Thread-10 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:07:28 - WARNING :: CP Server Thread-10 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:07:28 - WARNING :: CP Server Thread-10 : Unable to retrieve data for get_current_activity_header.
	2017-06-29 17:07:28 - WARNING :: CP Server Thread-9 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host
	2017-06-29 17:07:28 - WARNING :: CP Server Thread-9 : PlexPy Pmsconnect :: Unable to parse XML for get_current_activity: 'NoneType' object has no attribute 'getElementsByTagName'.
	2017-06-29 17:07:28 - WARNING :: CP Server Thread-9 : Unable to retrieve data for get_activity.
	2017-06-29 17:07:43 - WARNING :: CP Server Thread-7 : Failed to access uri endpoint /status/sessions with error [Errno 10054] An existing connection was forcibly closed by the remote host

**Comment by JonnyWong16** [comment]
I need to see everything before that.

**Comment by AngryNoodlez** [comment]
Hi Jonny,

I've attached them here

[plexpy.log.1.txt](https://github.com/JonnyWong16/plexpy/files/1112981/plexpy.log.1.txt)

**Comment by JonnyWong16** [comment]
There's the line I was looking for:

```
2017-06-29 06:49:26 - WARNING :: Thread-19 : PlexPy PlexTV :: Unable to retrieve server URLs. Using user-defined value without SSL.
```

PlexPy cannot reach Plex.tv.

```
2017-06-29 06:49:26 - WARNING :: Thread-19 : Failed to access uri endpoint /api/resources?includeHttps=1. Status code 503
```

Can you visit https://plex.tv/api/resources?includeHttps=1 from the machine running PlexPy? From the command line (with something like `cURL`)? Make sure there are no proxies, VPNs, firewalls, etc. which are blocking it.

**Comment by AngryNoodlez** [comment]
Hi Jonny,

I'm running it on Windows, using cURL 7.53.1 I can get to it ok;

    C:\Scripts\cURL>curl https://plex.tv/api/resources?includeHttps=1
    <?xml version="1.0" encoding="UTF-8"?>
    <errors>
      <error>Please sign in.</error>
    </errors>

**Comment by JonnyWong16** [comment]
Have you restarted PlexPy yet?

**Comment by AngryNoodlez** [comment]
Hmm unfortunately so. I'll roll back to the previous version of Plex to make sure this is definitely since the update only, and nothing else is going on.

**Comment by AngryNoodlez** [comment]
Ok,

So it's not the plex update which has killed it... I've rolled back and still getting the same issues.

I'll do a fresh install of PlexPy and see how I get on.

~Mat

**Comment by AngryNoodlez** [comment]
Mm. Not sure what it could be now... Fresh install of PlexPy seems to be doing the same.

**Comment by seanvree** [comment]
Windows 10E X64 1703
PMS:   1.7.5.4035
PPY: 1.4.21 MASTER
Python Version:	2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:42:59) [MSC v.1500 32 bit (Intel)]

Settings:
I"m NOT using SSL in PlexPY
PMS settings are set to "preferred"
No firewall on the router or Server

Hey @JonnyWong16 , I'm getting this EXACT same problem now as well. 
Noticed this tonight when I was trying to Tshoot a completely diff issue with MatterMost (I talked to you about it on Gitter).  I usually use the local admin accnt to log on, so I tried with the PMS admin (redacted@hotmail.com) and sure enough, couldn't log in... So I went back thru the logs and sure enough, started happening three days ago.  



```
2017-07-02 04:24:28 - WARNING :: CP Server Thread-10 : PlexPy WebAuth :: Unable to retrieve Plex.tv server token for user 'redacted@hotmail.com'.
2017-07-02 04:24:29 - DEBUG   :: CP Server Thread-10 : PlexPy WebAuth :: Invalid login attempt from 'redacted@hotmail.com'.
```

Now I can manually plug in the Token from PMS, and it still doesn't work.  Now if I take that SAME token and use it with OMBI.  NO problems. I can log on to  OMBI using PLEX.tv auth...sooooo, it can't be a communication issue with plex.tv?

Also, I can get get a token automatically just fine:

```
017-07-02 05:09:46 - INFO    :: CP Server Thread-3 : PlexPy PlexTV :: Fetching a new Plex.tv token for PlexPy.
2017-07-02 05:09:47 - INFO    :: CP Server Thread-3 : PlexPy Config :: Writing configuration to file
2017-07-02 05:09:47 - INFO    :: CP Server Thread-3 : PlexPy PlexTV :: Updated Plex.tv token for PlexPy.
2017-07-02 05:09:48 - DEBUG   :: CP Server Thread-9 : PlexPy APIv2 :: API called with kwargs: {'cmd': u'get_activity', 'apikey': u'********39'}
2017-07-02 05:09:48 - DEBUG   :: CP Server Thread-9 : PlexPy APIv2 :: Cleaned kwargs: {}
```

Something is def weird, there's a bug somewhere, not sure if it's python, PMS, PlexPY or something else.

**Comment by seanvree** [comment]
Hey @AngryNoodlez  are you by chance using DDWRT?

**Comment by AngryNoodlez** [comment]
Hi Seanvree,

No, using PfSense. Though communication internally is perfect, so not sure what the issue could be.

**Comment by seanvree** [comment]
@AngryNoodlez - Yup, same here, all communication is fine on the LAN.  Heard of PFS, but not using it...does it by chance/or are you using DNSMasq?

**Comment by AngryNoodlez** [comment]
@Seanvree - Nope, I have DNS Forwarder disabled (DNSMasq on pfsense). 

I very much doubt this is DNS related though, as the hashed SSL URL is working correctly across all of my network zones, and as you pointed out; other services which authenticate with Plex securely aren't suffering the same problems.

**Comment by JonnyWong16** [comment]
Shut down PlexPy, open the `config.ini` file and delete the line `pms_uuid`, start PlexPy and fetch a new token.

**Comment by seanvree** [comment]
@JonnyWong16  @AngryNoodlez 

That seemed to have done the trick!

Thanks!

**Comment by AngryNoodlez** [comment]
I'm getting data when now on PlexPy - though the "Verify" option in the settings still fails.

Does yours now verify OK @Seanvree?

**Comment by seanvree** [comment]
@AngryNoodlez 

Yup, mine is good to go:

2017-07-02 09:29:24 - DEBUG   :: CP Server Thread-12 : IP Checker :: Resolved localhost to 127.0.0.1.

**Comment by micle546** [comment]
Following the steps provided here, I am able to receive data from my local plex server until approximately the time the plex server goes through the daily maintenance phase.

@seanvree I am using DNSMasq. What effect does this have on the plex/plexpy server connection?
